### PR TITLE
fix: 添加h264编码器检测，不存在时不支持mp4视频录制

### PR DESCRIPTION
**Pull request by feeengli**
添加h264编码器检测，不存在时不支持mp4视频录制

Log: 添加h264编码器检测，不存在时不支持mp4视频录制

### CODE REVIEW

**Comment by myk1343** [comment]
[是否满足兼容性要求] Y
[是否满足commit提交规范] Y
[是否满足编码规范] Y
[Review结论] Pass
[Fail原因] N/A

**Review by hundundadi — COMMENTED** [review]
[是否满足兼容性要求] Y
[是否满足commit提交规范] Y
[是否满足编码规范] Y
[Review结论] Pass
[Fail原因] N/A

**Review by myk1343 — APPROVED** [review]
[是否满足兼容性要求] Y
[是否满足commit提交规范] Y
[是否满足编码规范] Y
[Review结论] Pass
[Fail原因] N/A

File: src/main.cpp [hunk 1]
```diff
@@ -82,7 +82,24 @@ static bool CheckFFmpegEnv()
...
     QStringList list = dir.entryList(QStringList() << (QString("libavcodec") + "*"), QDir::NoDotAndDotDot | QDir::Files);
+
     if (list.contains("libavcodec.so") || list.contains("libavcodec.so.58")) {
+        QLibrary libavcodec;  //检查编码器是否存在
+        libavcodec.setFileName("libavcodec.so");
+        qDebug() << "编码库是否加载成功: "<< libavcodec.load();
+        typedef AVCodec *(*p_avcodec_find_encoder)(enum AVCodecID id);
+        p_avcodec_find_encoder m_avcodec_find_encoder = nullptr;
+        m_avcodec_find_encoder = reinterpret_cast<p_avcodec_find_encoder>(libavcodec.resolve("avcodec_find_encoder"));
+        AVCodec *pCodec = m_avcodec_find_encoder(AV_CODEC_ID_H264);
```
[the remark this answers is not in the document]
Comment (feeengli) [reply src/main.cpp]: 已修改

**Review by lzwind — APPROVED** [review]
/+1

**Pull request merged**